### DISCUSSION TITLE
remove stale bot

### DISCUSSION THREAD

**Discussion by skyfaller** [extrawurst/gitui#1002] [Ideas]
Please disable stale bot on GitHub. It makes the open or closed status of an issue meaningless, and increases GitHub notification noise.

My understanding is that if an issue is closed, it means:

- the problem has been fixed, or
- the problem cannot be fixed (e.g. upstream bug), or
- the maintainer does not want it to be fixed

If stale bot runs around closing bugs willy nilly, now the closed state doesn't mean much of anything. The difference between a closed bug and an open bug can just be happenstance, or the confluence of tangential factors such as:

- Was the nature of the bug obvious? How much discussion was required before everyone understood it? Longer discussions mean more time before stale bot triggers. I think obvious bugs may sometimes be higher priority to fix.
- How many annoying users who like to complain did this issue affect? This gives incentives to users who care about the bug to be whiny, rather than wait patiently. It also fills my GitHub notifications with stale bot and people defending against stale bot, which now that I think about it may be the core of my grudge against stale bot. I certainly comment on any stale botted issue I follow, purely because I hate stale bot, but this is a little like emailing a large email thread asking to be removed because you're getting too many email notifications. Now we all must suffer.
- Did the person with the time and energy to fix it show up in 3 months or 6 months? This is an arbitrary matter of timing, and says nothing about the bug itself.

In my last confrontation with stale bot, me and the other user who cared about the bug eventually found someone to work on fixing it... But the bug is still closed! https://github.com/yuin/goldmark/issues/180 "Closed" doesn't even mean that the bug is inactive, that nobody is working on it, because stale bot sucks meaning out of the universe!

From now on I will link to this issue in issues that stale bot marks as stale, rather than repeating myself.

**Comment by extrawurst** [comment]
I responded to this here already: https://github.com/extrawurst/gitui/issues/865#issuecomment-974557426

>  "Closed" doesn't even mean that the bug is inactive

let me assure you it does in this repo

**Comment by skyfaller** [comment]
Thanks for the explanation, I understand your approach to GitHub issues better now. It seems you are treating the issues as your personal to-do list, rather than a place for the community to find issues to work on (or share workarounds). You are responsible for a popular FOSS project that I personally rely on and enjoy, so if this works for you I have to respect your process, even if I find it frustrating.

If you must use this approach, I encourage you to rethink how you manage closed issues. When I browse through closed issues, I generally cannot tell which are closed because they have been fixed/implemented, and which may still be worth working on even if nobody is currently working on them. There are a few exceptions, such as the "wontfix" label which certainly clarifies things, but I don't see a "done" or "finished" tag for example. There is no "upstream" tag indicating the fix must be submitted elsewhere.

When I click on the "good first issue" tag, I see a bunch of closed bugs, but which ones might still be good first issues, and which are no longer good first issues because they've been fixed/implemented?

It seems if you want the closed bugs to be useful to the community, you'd have to do a lot more with tagging. Simply not closing these bugs seems simpler to me, but I guess it's just a question of where you want to focus your maintenance efforts, since I acknowledge that open issues can eventually be left behind by evolving software and need to be periodically checked to see if they're still relevant.

I will say that the current system seems hostile to bugs that aren't quickly picked up by someone who wants to fix them, and I don't know how to keep a bug I care about in the public eye without repeatedly leaving comments to defend against stale bot. If I had confidence that the bug would still be easily found by people who want to find it, I wouldn't feel compelled to create this noise. How can someone who has the time and skill to fix my issue find it once it is closed, among all of the closed bugs that are no longer relevant?

I also encourage you to think beyond this specific project, to the broader GitHub community... even if you are able to maintain a clear distinction between active and inactive bugs, e.g. by reopening bugs that someone is working on, many projects like the one I complained about do not succeed at this. In other words, many people may not consider it to be a reliable signal of what is actually being worked on by someone (especially if that someone is not a maintainer) once stale bot is in the mix.
